src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,101 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.AbstractTest;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.util.UserTestUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+
+class UserDaoImplTest extends AbstractTest {
+    private UserDao userDao;
+    private SessionFactory sessionFactory;
+
+    @BeforeEach
+    void setUp() {
+        sessionFactory = Mockito.spy(getSessionFactory());
+        userDao = new UserDaoImpl(sessionFactory);
+    }
+
+    @Test
+    void findByEmail_Ok() {
+        Session session = Mockito.mock(Session.class);
+        Mockito.when(sessionFactory.openSession()).thenReturn(session);
+        Query query = Mockito.mock(Query.class);
+        Mockito.when(session.createQuery(anyString(), eq(User.class)))
+                .thenReturn(query);
+        Mockito.when(query.setParameter(anyString(), eq(UserTestUtil.EMAIL)))
+                .thenReturn(query);
+        Mockito.when(query.uniqueResultOptional())
+                .thenReturn(Optional.of(UserTestUtil.getUserBob()));
+        userDao.findByEmail(UserTestUtil.EMAIL);
+    }
+
+    @Test
+    void findByEmailNoInDataBase_NotOk() {
+        Optional<User> user = userDao
+                .findByEmail(UserTestUtil.INCORRECT_EMAIL);
+        Assertions.assertTrue(user.isEmpty());
+    }
+
+    @Test
+    void findById_Ok() {
+        Session session = Mockito.mock(Session.class);
+        Mockito.when(sessionFactory.openSession()).thenReturn(session);
```

Comment:
Why do we need to mock sessionFactory? As we have our test sessionFactory with in memory db connection  in userDao so we can test userDao method without mocking? Fix in all places